lib/check-meta.nix

```diff
@@ -0,0 +1,200 @@
+let
+  lib = import ./default.nix;
+in rec {
+
+  # Extend a derivation with checks for brokenness, license, etc.
+  # Throw a descriptive error when the check fails; return derivaitonArg otherwise.
+  # Note: no dependencies are checked in this step.
+  addMetaChecks = { config, platform, meta, derivationArg, mkDerivationArg }: let
```

Comment:
Yeah that one is the odd one out. I'll probably end generalizing that, rather than renaming it, because it's not possible to enumerate every conceivable compatible platform.